lib/trema/controller.rb

```diff
@@ -159,7 +164,21 @@ def run(args)
         File.expand_path(File.join(Phut.socket_dir, "#{name}.ctl"))
       @drb = DRb::DRbServer.new 'drbunix:' + drb_socket_file, self
       maybe_send_handler :start, args
-      socket = TCPServer.open('<any>', @port_number)
+
+      if Controller.ctl_privkey == nil || Controller.ctl_cert == nil ||
+         Controller.ca_certs == nil
+        socket = TCPServer.open('<any>', @port_number)
+        logger.debug "TCP socket"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.